dvc/hash_info.py

```diff
@@ -41,3 +41,9 @@ def isdir(self):
         if not self:
             return False
         return self.value.endswith(HASH_DIR_SUFFIX)
+
+    def as_raw(self) -> "HashInfo":
+        assert self.value
+        return HashInfo(
+            self.name, self.value.replace(HASH_DIR_SUFFIX, ""), self.obj_name
```

Comment:
using replace it not really needed and might actually hide some bugs

```suggestion
            self.name, self.value.rsplit(HASH_DIR_SUFFIX)[0], self.obj_name
```